controllers/infrastructure/byomachine_controller.go

```diff
@@ -349,22 +350,34 @@ func (r *ByoMachineReconciler) ClusterToByoMachines(logger logr.Logger) handler.
 
 // setNodeProviderID patches the provider id to the node using
 // client pointing to workload cluster
-func (r *ByoMachineReconciler) setNodeProviderID(ctx context.Context, remoteClient client.Client, host *infrav1.ByoHost, providerID string) error {
+func (r *ByoMachineReconciler) setNodeProviderID(ctx context.Context, remoteClient client.Client, host *infrav1.ByoHost) (string, error) {
 	node := &corev1.Node{}
 	key := client.ObjectKey{Name: host.Name, Namespace: host.Namespace}
 	err := remoteClient.Get(ctx, key, node)
-
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	if node.Spec.ProviderID != "" {
+		var match bool
+		match, err = regexp.MatchString("byoh://"+host.Name+"/.+", node.Spec.ProviderID)
```

Comment:
Suggestion: No strong opinions but I find `fmt.Sprintf` more readable instead of concatenation. wdyt?

Reply:
ok.